backend/src/middlewares/users/edit-user.middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { ErrorMessage, editUserSchema } from '@autoline/shared';
+import { TypedRequest } from '@common/types/controller/controller';
+import { UserUpdateInput } from '@controllers/users/users.controller';
+import { prisma } from '@data/prisma-client';
+import { NextFunction, Response } from 'express';
+import httpStatus from 'http-status-codes';
+
+const editUserMiddleware = async (
+  req: TypedRequest<{ id: string }, UserUpdateInput>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    await editUserSchema.validate(req.body);
+    const existedUser = await prisma.user.findFirst({
+      where: {
+        id: {
+          not: req.params.id,
+        },
+        email: req.body.email,
+      },
+    });
+    if (existedUser) next(new Error('User with this email already exists!'));
+
+    next();
+  } catch (err) {
+    const { message } = err as ErrorMessage;
+    res.status(httpStatus.FORBIDDEN).json({ error: message });
+    next(err);
+  }
+};
```

Comment:
This should be in the service layer, not in the middleware

Reply:
All this code to the service layer?
Or may be `await editUserSchema.validate(req.body);` should be left here in middleware? I assume it.

Reply:
I've moved this part of code.